docs/source/user-guide/formatter_objects.rst

```diff
@@ -0,0 +1,77 @@
+
+In order to run a model and data-agnostic comprehensive testing suites, the data should be in an accepted format the checks can work with.
+The formatter objects provide an easy way to do that, and are an important part of the vision module.
+
+
+Types of formatters
+==========================
+Deepchecks requires three data formatters:
+
+- **Image Formatter** - responsible for formatting the input images to the accepted format
+- **Prediction Formatter** - responsible for formatting the output of the model to an accepted format (depends on the model type)
+- **Label Formatter** - responsible for formatting the targets to an accepted format (depends on the task type)
+
+In order to initialize a new formatter, a transformation function between the input format and the accepted format is required.
+
+.. _formatters__image_formatter:
+
+The Image Formatter
+=====================
+The image formatter is responsible for formatting the image data outputted from the data loader to the required format for check displays.
+In order to create an image formatter, you need to initialize an instance of the :class:`image_formatters.ImageFormatter` class,
+passing the transformation function.
+
+Task types formats
+=====================
+Each CV task type (e.g. classification, object detection, instance segmentation, etc.) has its own data formats.
+For example, in classification tasks a common format for the label is simply the label_id, and for the prediction is a probability vector with the probability for each class. In object detection, there are various formats like the COCO format, PASCAL VOC XML format, YOLO format, and more...
+
+Deepchecks accepts a specific data format for each task type and for some types, provides utility functions to convert between common formats to the accepted one.
+
+.. _formatters__classification_formatter:
+
+Classification format
+-------------------------
+
+Accepted format
+~~~~~~~~~~~~~~~~~~~~~
+In classification tasks, it is required to implement the following formatters:
+
+- **Label**  - should implement an instance of :class:`classification_formatters.ClassificationLabelFormatter`.
+- **Predictions** - should implement an instance of :class:`classification_formatters.ClassificationPredictionFormatter`.
+
+.. _formatters__object_detection_formatter:
+
+Object detection format
+-------------------------
+
+Accepted Format
+~~~~~~~~~~~~~~~~
+In object detection tasks, it is required to implement the following formatters:
+
+- **Label** - should implement an instance of :class:`detection_formatters.DetectionLabelFormatter`.
+
+- **Detections** - should implement an instance of :class:`detection_formatters.DetectionPredictionFormatter`.
+
+Converting from common formats
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+TBD (Not implemented yet)
```

Comment:
```suggestion
In order to support more use-cases with less lines of code, we have implemented built-in formatters for many of the popular formats. For object detection, these will include the following supported formats:
            - 'xyxy' - x1, y1, x2, y2 represent the upper left and lower right corners of the bounding box.
            - 'xywh' - x1, y1, w, h represent the upper left corner of the bounding box and its width and height.
            - 'cxcywh' - cx, cy, w, h represent the center of the bounding box and its width and height.
            - 'xyxyn' - x1, y1, x2, y2, n represent the upper left and lower right corners of the bounding box,
                        normalized to the image dimensions.
            - 'xywhn' - x1, y1, w, h, represent the upper left corner of the bounding box and its width and height,
                        normalized by the image dimensions.
            - 'cxcywhn' - (YOLO format) x, y, w, h, represent the center of the bounding box and its width and height,
                          normalized by the image dimensions.
If your use case is in one of the supported formats please reach out to us and we can send you the required function (pre-release) and save you the implementation, since this is still in development. Alternatively, if you think your format is common but is not listed here, please reach out and we will consider adding it (Of course, issues on github and contributions after we release are also welcome).
```

Reply:
btw I'm considering delaying this task until we actually see that there are people that request it, WDYT?

Reply:
I agree. we shouldn't have it